src/custom_modules/DAO_modules/payment.ts

```diff
@@ -42,6 +42,24 @@ async function getPaymentMethodsByIDs(paymentMethodIDs) {
   }
 }
 
+/**
+ * computes the transaction cost for a donation with given sum and paymentID
+ * @param paymentId The payment method used
+ * @param sum The donation amount
+ * @returns {number} The transaction cost
+ * */
+async function computeTransactionCost(sum, paymentId) {
```

Comment:
This should be unecesary, the transaction cost is set by the database automatically (a trigger on update and insert).

Reply:
I did try without setting transaction cost at all first, but it didn't change.
Maybe it's just triggered on insert?

If you fix it I'm happy to remove this code.

This line in prod_schema looks relevant: 
`SET NEW.transaction_cost = ((SELECT percentage_fee FROM Payment where ID = NEW.payment_ID LIMIT 1)/100)*NEW.sum_confirmed + (SELECT flat_fee FROM Payment where ID = NEW.payment_ID LIMIT 1);`